lib/index.js

```diff
@@ -3148,6 +3148,73 @@ Stream.prototype.reduce1 = function (f) {
 exposeMethod('reduce1');
 
 /**
+ * Reduces a stream, using a stream-returning function. This allows async
+ * reduction.
+ *
+ * If the iterator throws an error, the reduction stops and the resulting
+ * stream will emit that error instead of a value.
+ *
+ * @id flatReduce
+ * @section Transforms
+ * @name Stream.flatReduce(memo, iterator)
+ * @param memo - the initial state of the reduction
+ * @param {Function} iterator - the function which reduces the values
+ * @api public
+ *
+ * var add = function (a, b) {
+ *     return _([a + b]);
+ * };
+ *
+ * _([1, 2, 3, 4]).flatReduce(0, add)  // => 10
+ */
+
+Stream.prototype.flatReduce = function(z, f) {
+    return this.reduce(_([z]), function(a, x) {
+        return a.flatMap(function(y) {
```

Comment:
I went with this implementation for clarity's sake and/or cribbing from Haskell's `foldM`.